### ISSUE TITLE
Campaign Managers

### ISSUE THREAD

**Issue by stevejgordon** [HTBox/allReady#1802]
Initial stub issue around requirement for more granular authorisation levels for allReady after discussions with @schuback

Campaign Managers should be able to perform the following actions within the scope of a specific campaign

- create events 
- create tasks 
- create itineraries 
- create manual requests 
- edit campaign details 
- close out task assignments 
- send messaging to all campaign levels

A user will be invited by an org admin to become a campaign admin

We will need to include a way or storing details of which users are campaign admins in our database. We will also need to include some changes within our authorisation code to ensure we allow the correct access to campaigns for the user's who should have it.

I will try to add some additional details / issues to this story soon.

**Comment by stevejgordon** [comment]
Working on a PR to enable easier authorization logic to support this "role"

**Comment by c0g1t8** [comment]
I'm going to take this one if no one minds.  I've got a lot practice with #1803.

**Comment by c0g1t8** [comment]
@stevejgordon @tonysurma 

I've got the _enable campaign manager_ part done but it's dependent on #1803.  I'll create a PR once #1803 is merged.

@stevejgordon In making these changes, it feels like there should be an `IAuthorizableOrganization `interface with associated support classes for consistency purposes.  I ended applying an `[Authorize("OrgAdmin")]` attribute to some methods.  Thoughts?

**Comment by stevejgordon** [comment]
> it feels like there should be an IAuthorizableOrganizationinterface with associated support classes for consistency purposes

@c0g1t8 Yep, that's my thought too. It makes sense to have a standard approach.

**Comment by c0g1t8** [comment]
@stevejgordon I'll take a crack at `IAuthorizableOrganization`
